src/dbnode/storage/series/buffer.go

```diff
@@ -114,13 +114,55 @@ type bufferStats struct {
 
 type bufferTickResult struct {
 	mergedOutOfOrderBlocks int
-	evictedBuckets         int
+	evictedBucketTimes     times
+}
+
+// times is a struct that holds an unknown number of times. This is used to
+// to avoid heap allocations as much as possible by trying to not allocate a
+// slice of times. To do this, `timesArraySize` needs to be strategically sized
+// such that for the vast majority of the time, the internal array can hold all
+// the times required so that `slice` is nil.
+type times struct {
```

Comment:
maybe add a comment to explain that this should only be used via the methods provided

Reply:
naming nit: `times` is really vague. pick something that explains what its for. like `optimisedTimes` or something?

Reply:
do you want to add tests to ensure we're not allocating unless necessary? (can test this by seeing `cap(times.slice)` isn't set unless arrIdx is >= timesArraySize)

Reply:
Will add a comment 👍

I'll call it `evictedTimes` since it should really only be used for this case since the array size should be tuned specifically for this.

Will add tests specifically for this struct.